src/Providers/Http/DTO/Request.php

```diff
@@ -0,0 +1,280 @@
+<?php
+
+declare(strict_types=1);
+
+namespace WordPress\AiClient\Providers\Http\DTO;
+
+use InvalidArgumentException;
+use WordPress\AiClient\Common\AbstractDataTransferObject;
+use WordPress\AiClient\Providers\Http\Enums\HttpMethodEnum;
+
+/**
+ * Represents an HTTP request.
+ *
+ * This class encapsulates HTTP request data that can be converted
+ * to PSR-7 requests by the HTTP transporter.
+ *
+ * @since n.e.x.t
+ *
+ * @phpstan-type RequestArrayShape array{
+ *     method: string,
+ *     uri: string,
+ *     headers: array<string, string|list<string>>,
+ *     body?: string|null
+ * }
+ *
+ * @extends AbstractDataTransferObject<RequestArrayShape>
+ */
+class Request extends AbstractDataTransferObject
+{
+    public const KEY_METHOD = 'method';
+    public const KEY_URI = 'uri';
+    public const KEY_HEADERS = 'headers';
+    public const KEY_BODY = 'body';
+
+    /**
+     * @var HttpMethodEnum The HTTP method.
+     */
+    protected HttpMethodEnum $method;
+
+    /**
+     * @var string The request URI.
+     */
+    protected string $uri;
+
+    /**
+     * @var array<string, string|list<string>> The request headers.
+     */
+    protected array $headers;
+
+    /**
+     * @var string|null The request body.
+     */
+    protected ?string $body;
+
+    /**
+     * Constructor.
+     *
+     * @since n.e.x.t
+     *
+     * @param HttpMethodEnum|string $method The HTTP method.
```

Comment:
Why do we here allow either when in other internal code we always mandate passing the enum itself? I think this needs to be changed for consistency.

Reply:
Good call! This cleans things up! Resolved in 53ccb48